lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n), because we have an array of chars, and we store the char counts in a hash.
 def is_palindrome(string)
```

Comment:
I think this solution ensures that the string **can** be a palindrome if you re-arrange the letters, not that it is one now.  

You would need to check with a loop through the string.